File: app/src/main/java/no/joharei/flixr/api/FlickrApiContainer.java
package no.joharei.flixr.api;


import no.joharei.flixr.api.models.Photos;
import no.joharei.flixr.api.models.Photosets;
import no.joharei.flixr.login.models.User;
import no.joharei.flixr.mainpage.models.Contacts;
import rx.Observable;

public interface FlickrApiContainer {
    Observable<User> getUserDetails();

    Observable<Photosets> getPhotosets(String userId);

    Observable<Photos> getPhotos(long photosetId, String userId);

    Observable<Contacts> getContacts();
}

File: app/src/main/java/no/joharei/flixr/api/FlickrApiContainer.kt
package no.joharei.flixr.api


import io.reactivex.Flowable
import io.reactivex.Observable
import no.joharei.flixr.api.models.Photos
import no.joharei.flixr.api.models.Photosets
import no.joharei.flixr.login.models.User
import no.joharei.flixr.mainpage.models.Contacts

interface FlickrApiContainer {
    fun getUserDetails(): Observable<User>

    fun getPhotosets(userId: String?): Flowable<Photosets>

    fun getPhotos(photosetId: Long, userId: String?): Observable<Photos>

    fun getContacts(): Observable<Contacts>
}
